MaskRCNN/dataset.py

```diff
@@ -54,24 +57,43 @@ def print_coco_metrics(self, json_file):
         """
         from pycocotools.cocoeval import COCOeval
         ret = {}
-        cocoDt = self.coco.loadRes(json_file)
-        cocoEval = COCOeval(self.coco, cocoDt, 'bbox')
-        cocoEval.evaluate()
-        cocoEval.accumulate()
-        cocoEval.summarize()
         fields = ['IoU=0.5:0.95', 'IoU=0.5', 'IoU=0.75', 'small', 'medium', 'large']
-        for k in range(6):
-            ret['mAP(bbox)/' + fields[k]] = cocoEval.stats[k]
-
-        json_obj = json.load(open(json_file))
-        if len(json_obj) > 0 and 'segmentation' in json_obj[0]:
-            cocoEval = COCOeval(self.coco, cocoDt, 'segm')
+        try:
+            cocoDt = self.coco.loadRes(json_file)
+            cocoEval = COCOeval(self.coco, cocoDt, 'bbox')
             cocoEval.evaluate()
             cocoEval.accumulate()
             cocoEval.summarize()
+
             for k in range(6):
-                ret['mAP(segm)/' + fields[k]] = cocoEval.stats[k]
-        return ret
+                ret['mAP(bbox)/' + fields[k]] = cocoEval.stats[k]
+
+            json_obj = json.load(open(json_file))
```

Comment:
Can we move line 71&72 to 62 to get rid of the try/except?

Reply:
Fixed